code/modules/recycling/disposal.dm

```diff
@@ -103,62 +119,97 @@
 			return
 		if((S.allow_quick_empty || S.allow_quick_gather) && length(S.contents))
 			S.hide_from(user)
-			user.visible_message("[user] empties \the [S] into \the [src].", "You empty \the [S] into \the [src].")
+			user.visible_message(
+				"<span class='notice'>[user] empties [S] into the disposal unit.</span>",
+				"<span class='notice'>You empty [S] into disposal unit.</span>",
+				"<span class='notice'>You hear someone emptying something full of smaller objects out into a disposal unit.</span>"
+				)
 			for(var/obj/item/O in S.contents)
 				S.remove_from_storage(O, src)
 			S.update_icon() // For content-sensitive icons
 			update()
 			return
 
+	// Borg using their gripper to throw stuff away.
+	if(istype(I, /obj/item/gripper/))
+		var/obj/item/gripper/gripper = I
+		// Gripper is empty.
+		if(!gripper.gripped_item)
+			to_chat(user, "<span class='warning'>You're not holding anything!</span>")
+			return
+		gripper.gripped_item.forceMove(src)
+		user.visible_message(
+			"<span class='notice'>[user] places [gripper.gripped_item] into the disposal unit.</span>", 
+			"<span class='notice'>You place [gripper.gripped_item] into the disposal unit.</span>",
+			"<span class='notice'>You hear someone dropping something into a disposal unit.</span>"
+			)
+		return
+
+	// Someone has a mob in a grab.
 	var/obj/item/grab/G = I
-	if(istype(G))	// handle grabbed mob
-		if(ismob(G.affecting))
-			var/mob/GM = G.affecting
-			for(var/mob/V in viewers(usr))
-				V.show_message("[usr] starts putting [GM] into the disposal.", 3)
-			if(do_after(usr, 20, target = GM))
-				GM.forceMove(src)
-				for(var/mob/C in viewers(src))
-					C.show_message("<span class='warning'>[GM] has been placed in [src] by [user].</span>", 3)
-				qdel(G)
-				add_attack_logs(usr, GM, "Disposal'ed", !!GM.ckey ? null : ATKLOG_ALL)
+	if(istype(G))
+		// If there's not actually a mob in the grab, stop it. Get some help.
+		if(!ismob(G.affecting))
+			return
+		var/mob/GM = G.affecting
+		for(var/mob/V in viewers(user))
+			V.show_message("[user] starts putting [GM] into the disposal unit.", 3)
+		// Abort if the target manages to scurry away.
+		if(!do_after(user, 20, target = GM))
+			return
+		GM.forceMove(src)
+		for(var/mob/C in viewers(src))
+			C.show_message("<span class='warning'>[GM] has been placed in the disposal unit by [user].</span>", 3)
+		qdel(G)
+		update()
+		add_attack_logs(user, GM, "Disposal'ed", !!GM.ckey ? null : ATKLOG_ALL)
```

Comment:
what is this `!!` syntax
```suggestion
		add_attack_logs(user, GM, "Disposal'ed", !GM.ckey ? null : ATKLOG_ALL)
```
meant to be one `!` ?

Reply:
This line was actually like this when I got here, I don't know the syntax for it.

Reply:
> what is this `!!` syntax
> 
> meant to be one `!` ?

It turns the value into boolean. I have had mixed messages on if it should be there or not in coding chat.

Reply:
It should be there, if `ckey` is `null`, it'll display `null`, and if not, it'll be `ATKLOG_ALL`
Only one `!`